yunwoo/baekjoon/브루트포스/2231-분해합.js

```diff
@@ -0,0 +1,33 @@
+const readline = require("readline");
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const input = [];
+
+rl.on("line", function (line) {
+  input.push(line);
+}).on("close", function () {
+  // 풀이
+  const n = +input[0];
+  const constructors = [];
+
+  for (let i = n; i > i / 2; i--) {
+    const convertedNumber = String(i);
+    let currentValue = i;
+
+    for (let j = 0; j < convertedNumber.length; j++) {
+      currentValue += +convertedNumber[j];
+    }
+
+    if (currentValue === n) constructors.push(i);
+  }
+  console.log(constructors.length ? Math.min(...constructors) : 0);
```

Comment:
```suggestion
  const n = +input[0];
  const constructors = [];

  for (let i = n; i > i / 2; i--) {
    const currentStrValue = String(i);
    let currentValue = i;

    for (let j = 0; j < currentStrValue.length; j++) {
      currentValue += +currentStrValue[j];
    }

    if (currentValue === n) constructors.push(i);
  }
  console.log(constructors.length ? Math.min(...constructors) : 0);
```
- 와우 너무 깔끔한 풀이 잘봤습니다 i를 n부터 감소시키면서 순회하는데 i > i / 2조건이 되나보네요...!! 배워갑니다 🙇🏻‍♂️
- convertedNumber 변수가 .. 음.. 고민인데요 ㅋㅋ convertedNumber변수라고 하니까 이게 문자열로 변환하는건지 확 와닿지가 않네요..!! 더 좋은 변수명이 있으면 좋을 것 같다는 생각이 들었네요. 저는 currentValue와 묶어서 currentStrValue도 좋을 것 같다는 생각이 들었습니다! 👍

Reply:
- ㅎㅎ 분해합이 절반이하로 가지 않을 것 같아 이렇게 짜봤는데 감사합니다 👍👍
- 변수명이 매번 제일 어렵습니다.. 🥲 currentStrValue이 훨씬 좋은 것 같네요ㅎㅎ 코멘트 감사합니다🙇‍♂️